src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpRawlsDAO.scala

```diff
@@ -87,5 +97,25 @@ class HttpRawlsDAO( implicit val system: ActorSystem, implicit val executionCont
   private def getWorkspaceUrl(ns: String, name: String) = FireCloudConfig.Rawls.authUrl + FireCloudConfig.Rawls.workspacesPath + s"/%s/%s".format(ns, name)
   private def patchWorkspaceAclUrl(ns: String, name: String) = rawlsWorkspaceACLUrl.format(ns, name)
 
+  override def getRefreshTokenStatus(userInfo: UserInfo): Future[Option[DateTime]] = {
+    val pipeline = addCredentials(userInfo.accessToken) ~> sendReceive
+    pipeline(Get(RawlsDAO.pathToUrl(RawlsDAO.refreshTokenDatePath))) map { response =>
+      response.status match {
+        case OK =>
+          Some(DateTime.parse(unmarshal[RawlsTokenDate].apply(response).refreshTokenUpdatedDate))
```

Comment:
use Option instead of Some directly so you don't run the risk of getting a Some(null)